base/replutil.jl

```diff
@@ -205,7 +205,11 @@ function showerror(io::IO, ex::MethodError)
         print(io, "This may have arisen from a call to the constructor $construct_type(...),",
                   "\nsince type constructors fall back to convert methods.")
     end
-    show_method_candidates(io, ex)
+    try
+        show_method_candidates(io, ex)
+    catch
+        warn("Unable to show method candidates")
```

Comment:
Yeah, you're right. Especially since we're already in the middle of producing output for an error. Will fix.